src/Orleans/Core/GrainFactory.cs

```diff
@@ -172,13 +186,29 @@ public Task<TGrainObserverInterface> CreateObjectReference<TGrainObserverInterfa
 
             Delegate creator;
 
-            if (!referenceCreators.TryGetValue(interfaceType, out creator))
+            IGrainMethodInvoker invoker;
+            if (!this.invokers.TryGetValue(interfaceType, out invoker))
+            {
+                invoker = MakeInvoker(interfaceType);
+                if (invoker != null)
+                {
+                    this.invokers.TryAdd(interfaceType, invoker);
+                }
+            }
+
+            if (invoker != null)
+            {
+                return Cast<TGrainObserverInterface>(await GrainReference.CreateObjectReference(obj, invoker));
+            }
+
+            if (!this.referenceCreators.TryGetValue(interfaceType, out creator))
```

Comment:
Yes, that's a fallback case. If I link this codegen into ClientGenerator, then this case isn't needed - it's purely for back-compat.